test/vm_tests/builder/parsing/members_test.dart

```diff
@@ -0,0 +1,1090 @@
+            file = SourceFile.fromString(r'''
+            UiFactory<FooProps> Foo = _$Foo;
+
+            mixin FooProps on UiProps {}
+  
+            mixin FooState on UiState {}
+  
+            class FooComponent extends UiStatefulComponent<FooProps, FooState>{}
+            ''');
+            collector = ErrorCollector.callback(file,
+                onError: validateCallback, onWarning: validateCallback);
+
+            component.validate(componentVersion, collector);
+            expect(validateResults, [contains('Must extend UiComponent2, not UiComponent.')]);
+          });
+        });
+
+        // Group that ensures that iterates over the deprecated lifecycles and all
+        // versions of the boilerplate, checking for errors where necessary.
+        group('detects deprecated lifecycles correctly', () {
+          Map<String, String> legacyLifecycleMethodsMap = {
+            'componentWillReceiveProps': 'Use getDerivedStateFromProps instead.',
+            'componentWillMount': 'Use init instead.',
+            'componentWillUpdate': 'Use getSnapshotBeforeUpdate instead.',
+          };
+
+          // Loop over the deprecated lifecycle methods
+          legacyLifecycleMethodsMap.keys.forEach((lifecycle) => {
+                // Loop through every version of the boilerplate
+                for (final version in BoilerplateVersions.values)
+                  {
+                    test('$lifecycle with ${versionDescriptions[version]}', () {
+                      // Grab the boilerplate with the deprecated lifecycle method
+                      final componentString = getBoilerplateString(
+                          deprecatedLifecycleMethod: lifecycle, version: version);
+                      final members =
+                          BoilerplateMemberHelper.parseAndReturnMembers(componentString);
+                      final component = members.whereType<BoilerplateComponent>().first;
+                      final componentVersion = resolveVersion(members).version;
+                      file = SourceFile.fromString(componentString);
+                      collector = ErrorCollector.callback(file, onError: validateCallback);
+
+                      component.validate(componentVersion, collector);
+
+                      // Warnings should only be logged if the component is a Component2
+                      if (component.isComponent2(componentVersion)) {
+                        expect(validateResults, [
+                          contains(legacyLifecycleMethodsMap[lifecycle]),
+                        ]);
+                      } else {
+                        expect(validateResults, []);
+                      }
+                    })
+                  }
+              });
+        });
+      });
+    });
+
+    group('factory', () {
+      group(
+          'hasFactoryAnnotation correctly identifies when a factory whether or not an annotation is present',
```

Comment:
I was really writing great test descriptions that day 🤦‍♂ 
```suggestion
          'hasFactoryAnnotation correctly identifies whether or not an annotation is present on a factory',
```